hawc/apps/study/api.py

```diff
@@ -47,6 +55,27 @@ def types(self, request):
         study_types = self.model.STUDY_TYPE_FIELDS
         return Response(study_types)
 
+    def create(self, request):
+
+        data = request.data.dict() if isinstance(request.data, QueryDict) else request.data
+        reference_id = tryParseInt(data.pop("reference_id", -1), -1)
+
+        try:
+            reference = Reference.objects.get(id=reference_id)
+        except ObjectDoesNotExist:
+            raise ValidationError(f"Reference ID does not exist.")
+        data["assessment"] = reference.assessment.id
+
+        self.check_object_permissions(request, reference)
+
+        serializer_class = self.get_serializer_class()
+        serializer = serializer_class(data=data, context={"reference_id": reference_id})
+        serializer.is_valid(raise_exception=True)
+
+        created_study = models.Study.save_new_from_reference(reference, serializer.validated_data)
```

Comment:
this should go in the `create` method https://www.django-rest-framework.org/api-guide/serializers/#saving-instances